x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/job_creator/multi_metric_job_creator.ts

```diff
@@ -69,4 +71,60 @@ export class MultiMetricJobCreator extends JobCreator {
   public removeDetector(index: number) {
     this._removeDetector(index);
   }
+
+  // called externally to set the model memory limit based current detector configuration
+  public async calculateModelMemoryLimit() {
+    if (this._splitField === null) {
+      // not split field, use the default
+      this.modelMemoryLimit = DEFAULT_MODEL_MEMORY_LIMIT;
+    } else {
+      const fieldNames = this._detectors.map(d => d.field_name).filter(fn => fn !== undefined);
+      const { modelMemoryLimit } = await ml.calculateModelMemoryLimit({
+        indexPattern: this._indexPatternTitle,
+        splitFieldName: this._splitField.name,
+        query: this._datafeed_config.query,
+        fieldNames,
+        influencerNames: this._influencers,
+        timeFieldName: this._job_config.data_description.time_field,
+        earliestMs: this._start,
+        latestMs: this._end,
+      });
+
+      try {
+        if (this.modelMemoryLimit === null) {
+          this.modelMemoryLimit = modelMemoryLimit;
+        } else {
+          // To avoid overwriting a possible custom set model memory limit,
+          // it only gets set to the estimation if the current limit is either
+          // the default value or the value of the previous estimation.
+          // That's our best guess if the value hasn't been customized.
+          // It doesn't get it if the user intentionally for whatever reason (re)set
+          // the value to either the default or pervious estimate.
+          // Because the string based limit could contain e.g. MB/Mb/mb
+          // all strings get lower cased for comparison.
+          const currentModelMemoryLimit = this.modelMemoryLimit.toLowerCase();
+          const defaultModelMemoryLimit = DEFAULT_MODEL_MEMORY_LIMIT.toLowerCase();
+          if (
+            currentModelMemoryLimit === defaultModelMemoryLimit ||
+            currentModelMemoryLimit === this._lastEstimatedModelMemoryLimit
+          ) {
+            this.modelMemoryLimit = modelMemoryLimit;
+          }
+        }
+        this._lastEstimatedModelMemoryLimit = modelMemoryLimit.toLowerCase();
+      } catch (error) {
```

Comment:
Do we want to do something with this error - maybe log it?

Reply:
hmm, i'm actually not sure what the correct behaviour is here. i don't want to add notify code to this non-ui class and we're not supposed to be using `console`.
The main point of this catch is to make sure the default mml is used if anything goes wrong.
I'll add a `console.error` with a  eslint-disable-next-line no-console` comment